File: sujeong/week3/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int count = 0;
int result = 0;
int globalTarget;
vector<int> globalNumbers;

void dfs(int sum, int i){
    if(i == globalNumbers.size()-1){
        if(sum == globalTarget)
            count++;
        return;
    }
    dfs(sum + globalNumbers[i+1], i+1);
    dfs(sum - globalNumbers[i+1], i+1);
}

int solution(vector<int> numbers, int target) {
    globalNumbers = {numbers.begin(), numbers.end()};
    globalTarget = target;
    
    dfs(numbers[0], 0);
    dfs(-numbers[0], 0);
    return count;
}

File: sujeong/week3/타겟 넘버.swift
import Foundation

var count: Int = 0

func dfs(_ numbers: [Int], _ target: Int, _ sum: Int, _ i: Int){
    if i == numbers.count - 1 {
        if sum == target {
            count += 1
        }
        return
    }
    
    dfs(numbers, target, sum + numbers[i+1], i+1);
    dfs(numbers, target, sum - numbers[i+1], i+1);
}

func solution(_ numbers:[Int], _ target:Int) -> Int {
    dfs(numbers, target, numbers[0], 0)
    dfs(numbers, target, -numbers[0], 0)
    
    return count
}
